content/en/docs/reference/networking/virtual-ips.md

```diff
@@ -0,0 +1,327 @@
+in-memory locking). Kubernetes also uses controllers to check for invalid
+assignments (eg due to administrator intervention) and for cleaning up allocated
+IP addresses that are no longer used by any Services.
+
+#### IP address ranges for `type: ClusterIP` Services {#service-ip-static-sub-range}
+
+{{< feature-state for_k8s_version="v1.25" state="beta" >}}
+
+Kubernetes divides the `ClusterIP` range into two bands, based on
+the size of the configured `service-cluster-ip-range` by using the following formula
+`min(max(16, cidrSize / 16), 256)`. That formula paraphrases as _never less than 16 or
+more than 256, with a graduated step function between them_.
+
+Kubernetes prefers to allocate dynamic IP addresses to Services by choosing from the upper band,
+which means that if you want to assign a specific IP address to a `type: ClusterIP`
+Service, you should manually assign an IP address from the **lower** band. That approach
+reduces the risk of a conflict over allocation.
+
+If you disable the `ServiceIPStaticSubrange`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) then Kubernetes
+uses a single shared pool for both manually and dynamically assigned IP addresses,
+that are used for `type: ClusterIP` Services.
+
+
+### Service IP addresses {#ips-and-vips}
+
+Unlike Pod IP addresses, which actually route to a fixed destination,
+Service IPs are not actually answered by a single host.  Instead, kube-proxy
+uses iptables (packet processing logic in Linux) to define _virtual_ IP addresses
+which are transparently redirected as needed.  When clients connect to the
+VIP, their traffic is automatically transported to an appropriate endpoint.
+The environment variables and DNS for Services are actually populated in
+terms of the Service's virtual IP address (and port).
+
+kube-proxy supports three proxy modes&mdash;userspace, iptables and IPVS&mdash;which
+each operate slightly differently.
+
+#### Userspace
+
+{{< feature-state for_k8s_version="v1.23" state="deprecated" >}}
+
+As an example, consider the image processing application described above.
+When the backend Service is created, the Kubernetes master assigns a virtual
+IP address, for example 10.0.0.1.  Assuming the Service port is 1234, the
+Service is observed by all of the kube-proxy instances in the cluster.
+When a proxy sees a new Service, it opens a new random port, establishes an
+iptables redirect from the virtual IP address to this new port, and starts accepting
+connections on it.
+
+When a client connects to the Service's virtual IP address, the iptables
+rule kicks in, and redirects the packets to the proxy's own port.
+The "Service proxy" chooses a backend, and starts proxying traffic from the client to the backend.
+
+This means that Service owners can choose any port they want without risk of
+collision.  Clients can connect to an IP and port, without being aware
+of which Pods they are actually accessing.
+
+#### iptables
```

Comment:
This subsection (and the next subsection) should be merged to the previous subsections. Most of the text are just repeating the same logic again and again.

Since we are "migrating" the reference details anyway, and it is almost impossible to trace back how paragraphs are relocated, we may want to take this opportunity to consolidate all proxy mode contents and make the text flow.

Reply:
If some of my other PRs merge, I can shrink https://github.com/kubernetes/website/pull/30817 until it reaches a reviewable size, and then that PR (#30817) becomes a way to address this feedback.